remote-trait-object/src/context.rs

```diff
@@ -55,13 +55,38 @@ mod meta_service {
 }
 use meta_service::{MetaService, MetaServiceImpl};
 
+/// A configuration of a `remote-trait-object` context.
 #[derive(Clone, Debug)]
 pub struct Config {
-    /// This will be appended to the names of various threads spawned by RTO, for an easy debug.
+    /// A name that will be appended to the names of various threads spawned by `remote-trait-object`, for an easy debug.
+    ///
+    /// This can be helpful if you handle multiple contexts of `remote-trait-object`.
     pub name: String,
+
+    /// Number of the maximum of concurrent calls.
+    ///
+    /// Value of this doesn't have anything to do with the number of threads that would be spawned.
+    /// Having a large number of this wouldn't charge any cost except really small additional memory allocation.
+    ///
+    /// [`server_threads`]: ./struct.Config.html#field.server_threads
     pub call_slots: usize,
-    pub call_timeout: std::time::Duration,
 
+    /// A timeout for a remote method call.
+    ///
+    /// All remote method invocations through your remote object and delete requests (that happens when you drop a remote object)
+    /// will have this timeout. If it exceeds, it will cause an error.
+    ///
+    /// Use `None` for to wait indefinitely.
+    pub call_timeout: Option<std::time::Duration>,
+
+    /// A maximum number of services that this context can export.
+    pub maximum_services_num: usize,
```

Comment:
I think it's subtle to provide this as an option.

It may hard to predict what will be a good maximum service number. 
I don't know what will be a good `maximum_services_num` value when I implement Foundry modules. However small `maximum_services_num` may make a deadlock or panic. 

If we want to make a good library that is easy to use, it will be better the library manages these things in its code.

Reply:
My intention is make as many options as possible, for full control over the context. I think we can think about this later